crates/core/src/chunker.rs

```diff
@@ -165,7 +163,7 @@ impl<R: Read + Send> Iterator for ChunkIter<R> {
             self.pos += 1;
             self.rabin.slide(byte);
         }
-        self.size_hint -= vec.len();
+        self.size_hint = self.size_hint.saturating_sub(vec.len()); // size_hint can be too small!
```

Comment:
nice
do we want to also give an example of that happening here ? (like linux pseudo fs in /proc)

Reply:
I think the main use case is just that between `stat`ing a file and opening it, there could be another process which has modified the file - potentially making it larger..
But I would not add an explicit example; it is simply preventing an underflow here which just makes sense...